inc/db_mysql.php

```diff
@@ -788,7 +788,9 @@ function insert_query($table, $array)
 			}
 			else
 			{
-				$array[$field] = "'{$value}'";
+				$quoted_value = $this->quote_val($value);
+
+				$array[$field] = "{$quoted_value}";
```

Comment:
Guess Euan missed that and simply modified the original line. Will change that (and similar lines)